onnx/defs/tensor/defs.cc

```diff
@@ -170,6 +171,21 @@ Example 2:
             "Constrain input and output types to float tensors.");
 
 ONNX_OPERATOR_SCHEMA(Transpose)
+.ShapeInferenceFunction([](const Node* n, NodeOutTypes& out_types, NodeOutSizes& out_dims) {
+      auto inp = n->inputs()[0];
+      auto out = n->outputs()[0];
+      out_types.push_back(inp->hasElemType() ? inp->elemType() : ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED);
+      out_dims.emplace_back();
+      if (!inp->sizes().empty()) {
+        for (size_t i = 0; i < inp->sizes().size(); i++) {
+          size_t idx = n->hasAttribute(kperm)
+            ? n->is(kperm)[i]
+            : inp->sizes().size() - i - 1
+            ;
+          out_dims.back().push_back(inp->sizes()[idx]);
```

Comment:
`emplace_back` is never worse and sometimes better for perf than `push_back`, depending on how the `Dimension` type is implemented.